keps/sig-auth/20190806-serviceaccount-tokens.md

```diff
@@ -263,6 +264,35 @@ service account token on behalf of pods running on that node. The
 NodeRestriction admission controller will require that these tokens are pod
 bound.
 
+### Safe Adoption
+
+#### Blocker: In-cluster clients should reload fresh token
+
+- Summary: Bound service account tokens are time-bound. If in-cluster clients 
+do not load fresh token from projected volume, tokens would be rejected since 
+they are expired.
+
+- Rollout plan:
+  1. Modify admission-injected token source to request for token expiring at an 
+  uncommon value. (1 hour 10 seconds for example)
+  1. Modify kube apiserver TokenRequest API, providing an option 
+  `--mint-longer-token` to mint a token longer than requested if expiration 
+  requested is the uncommon value set in step i (+30 days for example), while 
+  putting original expiration requested to a new field `kubernetes.io/warn-after` 
+  in token. The step ensures clients adopting bound service account token with 
+  auto-injection would get warning but not breaking if failed to reload fresh token.
+  1. Add metrics into kube apiserver to collect usage of:
+     * stale_token: current time after `kubernetes.io/warn-after` field
+     * legacy_token: audience not available in token
+     * valid_token
```

Comment:
cc @logicalhan on suggestions for name/format on these

Reply:
We could review metrics format in PR later.